gnocchi/cli/injector.py

```diff
@@ -0,0 +1,121 @@
+                   default="low"),
+        cfg.IntOpt("--interval",
+                   help="Interval to sleep between metrics sending."),
+        cfg.BoolOpt("--process", default=False,
+                    help="Process the ingested measures."),
+    ])
+    return _inject(service.prepare_service(conf=conf, log_to_std=True),
+                   metrics=conf.metrics,
+                   measures=conf.measures,
+                   archive_policy_name=conf.archive_policy_name,
+                   process=conf.process,
+                   interval=conf.interval)
+
+
+def _inject_from_conf(conf,
+                      metrics, measures, archive_policy_name="low",
+                      process=False, interval=None):
+    inc = incoming.get_driver(conf)
+    coord = metricd.get_coordinator_and_start(str(uuid.uuid4()),
+                                              conf.coordination_url)
+    store = storage.get_driver(conf)
+    idx = indexer.get_driver(conf)
+    return _inject(inc, coord, store, idx,
+                   metrics, measures, archive_policy_name, process, interval)
+
+
+def _inject(inc, coord, store, idx,
+            metrics, measures, archive_policy_name="low", process=False,
+            interval=None):
+    LOG.info("Creating %d metrics", metrics)
+    with utils.StopWatch() as sw:
+        metrics = [
+            idx.create_metric(uuid.uuid4(), "admin",
+                              archive_policy_name).id
+            for _ in range(metrics)
+        ]
+    LOG.info("Created %d metrics in %.2fs", metrics, sw.elapsed())
+
+    LOG.info("Generating %d measures per metric for %d metrics… ",
+             measures, metrics)
+    now = numpy.datetime64(utils.utcnow())
+    with utils.StopWatch() as sw:
+        measures = {
+            m_id: [incoming.Measure(
+                now + numpy.timedelta64(seconds=s),
+                random.randint(-999999, 999999)) for s in range(measures)]
+            for m_id in metrics
+        }
+    LOG.info("… done in %.2fs", sw.elapsed())
+
+    interval_timer = utils.StopWatch().start()
+
+    while True:
+        interval_timer.reset()
+        with utils.StopWatch() as sw:
+            inc.add_measures_batch(measures)
+        total_measures = sum(map(len, measures.values()))
+        LOG.info("Pushed %d measures in %.2fs",
+                 total_measures,
+                 sw.elapsed())
+
+        if process:
+            c = chef.Chef(coord, inc, idx, store)
+
+            with utils.StopWatch() as sw:
+                for s in inc.iter_sacks():
+                    c.process_new_measures_for_sack(s, blocking=True)
+            LOG.info("Processed %d sacks in %.2fs",
+                     inc.NUM_SACKS, sw.elapsed())
+            LOG.info("Speed: %.2f measures/s",
+                     float(total_measures) / sw.elapsed())
+
+        if interval is None:
+            break
+        time.sleep(max(0, interval - interval_timer.elapsed()))
```

Comment:
does this loop forever until you ctrl+c?

Reply:
If you set an interval, yes.